### PR TITLE
fix: Only report http2 support if the http2 feature is enabled

### PR DESCRIPTION
**Pull request by Marwes**
Fixes https://github.com/ctz/hyper-rustls/issues/143

### CODE REVIEW

File: src/connector.rs [hunk 1]
```diff
@@ -61,7 +61,15 @@ impl HttpsConnector<HttpConnector> {
...
-        config.alpn_protocols = vec![b"h2".to_vec(), b"http/1.1".to_vec()];
+        #[cfg(feature = "http2")]
+        {
+            config.alpn_protocols = vec![b"h2".to_vec(), b"http/1.1".to_vec()];
+        }
+        #[cfg(not(feature = "http2"))]
+        {
+            config.alpn_protocols = vec![b"http/1.1".to_vec()];
+        }
+
```
Comment (g2p) [inline src/connector.rs:72]: I think alpn_protocols should just be omitted if it only contains "http/1.1".

**Comment by Arnavion** [comment]
`hyper` allows toggling `http1` and `http2` features independently. So you also need to provide a knob for the ALPN negotiation to only include `"h2"` and not `"http/1.1"` for the case where the client only supports HTTP/2 and not /1.1

**Comment by paolobarbolini** [comment]
Any updates on this?

**Comment by djc** [comment]
This makes sense to me, thanks!

**Pull request merged**